### ISSUE TITLE
mp3 files aren't cached

### ISSUE THREAD

**Issue by tacman** [Spomky-Labs/phpwa#81]
### Version(s) affected

1.0

### Description

I have a [white noise app](https://noise.survos.com/) that almost works offline, but fails because the mp3 file isn't cached.  

Repo at https://github.com/survos-sites/noise, it's basically a Symfony version of https://github.com/wes-goulet/background-noise.  

### How to reproduce

```bash
git clone git@github.com:survos-sites/noise.git && cd noise
composer install
symfony server:start -d
symfony open:local
```
Press the play button (or space) to start the noise.

# Offline

Turn off the network and refresh the page.  No mp3.  Actually, lots of red in the debugger, but I think most of those are because of the assets not being CacheOnly.

### Possible Solution

```php
    #[SerializedName('image_regex')]
    public string $imageRegex = '/\.(ico|png|jpe?g|gif|svg|webp|bmp)$/';
```

rename imageRegex to mediaRegex, and include .mp3 and other audio/video formats.

Or even better, install MimeTypes.

```php
use Symfony\Component\Mime\MimeTypes;

$mimeTypes = new MimeTypes();
$mimeType = $mimeTypes->guessMimeType('/some/path/to/image.gif');
```

https://symfony.com/doc/current/components/mime.html#guessing-the-mime-type

**Comment by Spomky** [comment]
Hi,

Not all files are cached, especially audio/video files that could be heavy.
To me, there is no need to change the name or the regex. You can add a dedicated Workbox rule for this specific use case

```js
// assets/sw.js


// *** Workbox Bundle rules ***
//WORKBOX_IMPORT_PLACEHOLDER
//STANDARD_RULES_PLACEHOLDER
//PRECACHING_PLACEHOLDER
//WARM_CACHE_URLS_PLACEHOLDER
//OFFLINE_FALLBACK_PLACEHOLDER
//WIDGETS_PLACEHOLDER

const audioCacheStrategy = new workbox.strategies.CacheFirst({
  cacheName: 'audio',
  plugins: [
    new workbox.cacheableResponse.CacheableResponsePlugin({statuses: [0, 200]}),
    new workbox.expiration.ExpirationPlugin({
      maxEntries: 10, // Adapt this line if needed
      maxAgeSeconds: 365 * 24 * 60 * 60 // Adapt this line if needed
    }),
  ],
});

workbox.routing.registerRoute(
  ({url}) => url.pathname.endsWith('.mp3'),
  audioCacheStrategy
);

// Optional: this will store the file when the user arrives on the application
// audioFiles contains the list of URLs
const audioFiles = [
  '/audio/startup.mp3',
  '/audio/woop.mp3',
  '/audio/ending.mp3'
];
self.addEventListener('install', event => {
  const done = audioFiles.map(
    path =>
      audioCacheStrategy.handleAll({
        event,
        request: new Request(path),
      })[1]
  );

  event.waitUntil(Promise.all(done));
});
```

**Comment by tacman** [comment]
```js
const audioFiles = [
  '/audio/startup.mp3',
  '/audio/woop.mp3',
  '/audio/ending.mp3'
];
```

Does this handle the cached filenames with the hash tag?

**Comment by tacman** [comment]
Hmm, still not working.  Even hacking it the image regex, I see that it's being cached now, but still doesn't work offline.

https://noise.survos.com

**Comment by tacman** [comment]
Fixed with static_regex and 1.0.4
